beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogFixedTermWithVersionedCatalogs.java

```diff
@@ -123,6 +134,7 @@ public void testVersionedCatalogsEffectiveDateAfterEndOfFixedTerm() throws Excep
         busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
         clock.addMonths(1); //2021-04-01
         assertListenerStatus();
+        //Fixed term duration of 100 days ends on 2021-04-11
         invoiceChecker.checkInvoice(account.getId(), 4, testCallContext, new ExpectedInvoiceItemCheck(new LocalDate(2021, 04, 01), new LocalDate(2021, 04, 11), InvoiceItemType.RECURRING, new BigDecimal("13.33")));
         
         checkNoMoreInvoiceToGenerate(account.getId(), testCallContext);
```

Comment:
Just a general query, the `testVersionedCatalogsEffectiveDateAfterEndOfFixedTerm`  tests the `plumber-insurance-monthly2` plan. In this plan, the `effectiveDateForExistingSubscriptions` is `2021-04-12` which is at the end of the fixed-term duration. Wondering how such a catalog does not make sense. I mean, what use would a plan whose `effectiveDate`  is after the end of the fixed-term duration be?

Reply:
2 observations:
1. You wrote the test `testVersionedCatalogsEffectiveDateAfterEndOfFixedTerm`, so you tell me :-) ?
2.  `In this plan, the effectiveDateForExistingSubscriptions is 2021-04-12 which is at the end of the fixed-term duration.` -> It depends when you start your subscription - in your test, this is indeed useless, but I thought that's what you wanted to verify. But if you had started the subscription later, things could have been different, e.g your previous scenario `testVersionedCatalogsEffectiveDateBetweenFixedTerm `

Reply:
Agreed, makes sense. I wrote the test because I found that in order for the `https://github.com/killbill/killbill/blob/fix-attempt-fix-term/subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java#L674` to be executed, the `effectiveDateForExistingSubscriptions` should be after the fixed term period. I guess what I was really asking is what real-world situation would lead to the code at `https://github.com/killbill/killbill/blob/fix-attempt-fix-term/subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java#L674` to be executed since such a plan (which we agree is useless) would never be used in the real world. Anyway never mind, it's not really relevant here, so don't want to take your time :)

Reply:
Yes, I understand your point, you are correct: If the FIXEDTERM subscription expires prior the catalog change happens, returning such catalog change billing event is a no-op, it does not hurt, but also does not serve any purpose as it is never used.